src/tree.rs

```diff
@@ -22,12 +22,15 @@ use std::sync::atomic::{AtomicU32, Ordering};
 /// The tree has four sections:
 ///
 /// - `u32` ID
+/// // One thing to ponder is whether text *should* be a part of a tree?
+/// // If the tree stores *only* the ranges, then it becomes naturally generic
+/// // over the representation of text, and one might use a rope or a str.
```

Comment:
Hmm, that is a good point, though I’d imagine the vast vast majority of use-cases would be `str`; the syntax tree is immutable, so a rope is unnecessary. Because of this I would think making the tree generic over how text is represented adds a lot of complexity for little benefit.